x-pack/plugin/mapper-counted-keyword/src/test/java/org/elasticsearch/xpack/countedkeyword/CountedKeywordFieldMapperTests.java

```diff
@@ -118,7 +119,8 @@ public void testSyntheticSourceManyNullValue() throws IOException {
     }
 
     public void testSyntheticSourceIndexLevelKeepArrays() throws IOException {
-        SyntheticSourceExample example = syntheticSourceSupportForKeepTests(shouldUseIgnoreMalformed()).example(1);
+        SyntheticSourceExample example = syntheticSourceSupportForKeepTests(shouldUseIgnoreMalformed(), Mapper.SourceKeepMode.ARRAYS)
```

Comment:
Why is this change needed?

Reply:
The `NumberSyntheticSourceSupportForKeepTests` needs to know whether to expect the output value to lose precision or not. I.e., if the input is `1.5` to a field that is `type: long`, should the synthetic source produced have the value `1.5` or `1`? If we're in `source_keep_mode: all`, the exact field value is stored in ignored source, so no precision is lost (`1.5`). If we're in `source_keep_mode: arrays` or `source_keep_mode: none`, there is that loss of precision (`1`).

As such, I had to update the method signature for `MapperTestCase#syntheticSourceForKeepTests` to include the source keep mode. This means I also had to update every override and every call site.